sky/jobs/utils.py

```diff
@@ -516,17 +623,20 @@ def stream_logs(job_id: Optional[int],
         log_lib.tail_logs(job_id=job_id, log_dir=log_dir, follow=follow)
         return ''
 
-    if job_id is None:
-        assert job_name is not None
-        job_ids = managed_job_state.get_nonterminal_job_ids_by_name(job_name)
-        if len(job_ids) == 0:
-            return f'No running managed job found with name {job_name!r}.'
-        if len(job_ids) > 1:
-            raise ValueError(
-                f'Multiple running jobs found with name {job_name!r}.')
-        job_id = job_ids[0]
+    else:
+        if job_id is None:
+            assert job_name is not None
+            job_ids = managed_job_state.get_nonterminal_job_ids_by_name(
+                job_name)
+            if len(job_ids) == 0:
+                return f'No running managed job found with name {job_name!r}.'
+            if len(job_ids) > 1:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(
+                        f'Multiple running jobs found with name {job_name!r}.')
+            job_id = job_ids[0]
 
-    return stream_logs_by_id(job_id, follow)
+        return stream_logs_by_id(job_id, task_id, follow)
```

Comment:
Lets remove the `else` and revert the indents? We should reduce indent if possible.

Reply:
Thank you for the input! I agree that too much indentation isn’t ideal, but in this case, a couple of levels seem fine. Early exits work well for handling specific errors, while here, both cases are part of the function’s main flow.